### PR TITLE
Fix assertion failure when disconnecting the Ethernet cable (NXP1060 -  NetworkInterface.c)

### PR DESCRIPTION
**Pull request by microcris**
The changes that I'm proposing are to make the driver to work in case of cable disconnection. 

Description
-----------
In prvEMACHandlerTask, when xSendEventStructToIPTask, we have to pass network interface or it will fail the assertion in xSendEventStructToIPTask "configASSERT( pxInterface != NULL )".

In pxNXP1064_NetworkInterfaceInitialise, if xWaitPHY fails (for instance, the cable is not connected), the switch case will break without doing nothing and the prvEMACHandlerTask will be blocked waiting for the notification. We have to unblock it to try again in case of failure in xEMAC_WaitPHY.

Test Steps
-----------
With the original code, connect the Ethernet  cable, let the DHCP make what it has to make, disconnect the cable and you will get the following message:
Link down!
ASSERT ERROR " pxInterface != ((void *)0) ": file "../FreeRTOS_TCP/Source/FreeRTOS_IP_Utils.c" Line "823" function name "prvProcessNetworkDownEvent" 

Apply the fix and you will get:
Link down!
Network Down
Configuration successful.
Waiting for link to go up and auto-negotiation to complete.
Failed to get the link up.
Link down!
Network Down
Configuration successful.
Waiting for link to go up and auto-negotiation to complete.
Link up.Waiting for auto-negotiation to complete.
Auto-negotiation complete.

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [ ] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by microcris** [comment]
> Hi @microcris, Thanks for sharing the fix. Can you also share how you validated the same fix and the various tests that you did validate the functionalities on NXP1060.

Hello!

I'm running FreeRTOS and FreeRTOS-PLUS-TCP on a custom board based on NXP RT1064.
In this I'm running a DHCP client and a TCP socket server.

The board is getting a correct IP from a DHCP server, I'm able to connect a socket client that is running on a computer and I'm able to correctly exchange data between the computer and the board.

Is this enough? 

In this particular case, I don't know if I have to run all the tests that are in the folder "test".

File: source/portable/NetworkInterface/NXP1060/NetworkInterface.c [hunk 1]
```diff
@@ -568,13 +573,13 @@ static void prvEMACHandlerTask( void * parameter )
                         break;
 
                     case kStatus_ENET_RxFrameError: /* Received an error frame.  Read & drop it */
-                        PRINTF( "RX Receive Error\n" );
+                        FreeRTOS_printf( "RX Receive Error\n" );
```
Comment (ActoryOu) [inline source/portable/NetworkInterface/NXP1060/NetworkInterface.c:576]:
```suggestion
                        FreeRTOS_printf( ( "RX Receive Error\n" ) );
```

File: source/portable/NetworkInterface/NXP1060/NetworkInterface.c [hunk 2]
```diff
@@ -568,13 +573,13 @@ static void prvEMACHandlerTask( void * parameter )
...
                         ENET_ReadFrame( ethernetifLocal->base, &( ethernetifLocal->handle ), NULL, 0, 0, NULL );
                         /* Not sure if a trace is required.  The MAC had an error and needed to dump bytes */
                         break;
 
                     default:
-                        PRINTF( "RX Receive default" );
+                        FreeRTOS_printf( "RX Receive default" );
```
Comment (ActoryOu) [inline source/portable/NetworkInterface/NXP1060/NetworkInterface.c:582]:
```suggestion
                        FreeRTOS_printf( ( "RX Receive default" ) );
```

**Review by ActoryOu — CHANGES_REQUESTED** [review]
Hi @microcris,
Sorry for late response. And thanks for your contribution!

Overall, the change works perfectly. I can disconnect/re-connect network successfully by unplug/re-plug ethernet cable.

There are two minor changes required. The format of FreeRTOS_printf is different with your change and it causes compilation errors. Please help to take a look.

Thanks!

**Review by ActoryOu — APPROVED** [review]

**Review by shubnil — APPROVED** [review]

**Pull request merged**